webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteActions.js

```diff
@@ -0,0 +1,75 @@
+import URI from 'urijs';
+import API from '../../API';
+import { STATUS } from '../../constants';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+  TRIGGERS,
+} from './AutoCompleteConstants';
+import { objectDeepTrim } from '../../common/helpers';
+
+export const getResults =
+(path, searchQuery, controller, trigger) => (dispatch) => {
+  const loadNextResults = trigger === TRIGGERS.ITEM_SELECT ? '+' : '';
+  const url = `${path}${URI.encodeQuery(searchQuery)}${loadNextResults}`;
+
+  dispatch({
+    type: AUTO_COMPLETE_REQUEST,
+    payload: {
+      controller,
+      searchQuery,
+      status: STATUS.PENDING,
+      trigger,
+    },
+  });
+
+  return API.get(url)
+    .then(({ data }) => {
+      const { error } = data[0] || {};
+      if (error) {
+        throw error;
+      }
+      const results = data.map(result => objectDeepTrim(result));
+      return dispatch({
+        type: AUTO_COMPLETE_SUCCESS,
+        payload: {
+          controller,
+          results,
+          searchQuery,
+          status: STATUS.RESOLVED,
+          trigger,
+        },
+      });
+    })
+    .catch(error => dispatch({
+      type: AUTO_COMPLETE_FAILURE,
+      payload: {
+        results: [],
+        error: error.message || error,
+        status: STATUS.ERROR,
+      },
+    }));
+};
+
+export const resetData = controller => dispatch =>
+  dispatch({
+    type: AUTO_COMPLETE_RESET,
+    payload: { controller },
+    error: null,
+  });
+
+
+export const initialUpdate = (searchQuery, controller) => dispatch =>
```

Comment:
nitpick: extra whitespace above this line, in general I would recommend to use prettier / eslint to avoid many of these issues.